### PR TITLE
Thread Safety Change for MJPG V4L2

### LINKED ISSUE #387: Motion mixing images with two cameras
Hello,
I am using motion 4.0.1 with two USBcameras in a raspberry pi 3. I am using thread files. Motion sometimes mixes both images in a weird way. I attach an example, my conf and the log when it happens. When I reboot I almost always get mixed images. I installed motion from Github.

Thanks for the help. 

https://gist.github.com/drojasug/117b46b8d3aa58533da0133dfc3ee655

### PR DESCRIPTION
**Pull request by Mr-DaveDev**
This commit revises the jpeg utility for thread safety by moving the declares for the buf0-2 and chr1-2 variables into the associated functions rather than being at the module level hidden in the middle
of the code.

For the decode_jpeg_gray_raw this routine appears to have been copied from the decode_jpeg_raw when it was initially created.  When it was copied, it did not however remove all the code that was related to the Cr and Cb segments which do not exist in grey images.  Since the compiler now reports these as unused, this commit removes them to clean up the code.

Closes #387

### CODE REVIEW

File: jpegutils.c [hunk 1]
```diff
@@ -452,6 +444,13 @@ int decode_jpeg_raw (unsigned char *jpeg_data, int len,
     int i, xsl, xsc, xs, hdown;
     unsigned int x, y = 0, vsf[3], xd;
 
+    unsigned char buf0[16][MAX_LUMA_WIDTH];
+    unsigned char buf1[8][MAX_CHROMA_WIDTH];
+    unsigned char buf2[8][MAX_CHROMA_WIDTH];
+    unsigned char chr1[8][MAX_CHROMA_WIDTH];
+    unsigned char chr2[8][MAX_CHROMA_WIDTH];
```
Comment (jogu) [inline jpegutils.c]: If I did my maths right, we're putting 64K of data onto the stack. That seems like quite a large chunk. I think I've generally worked to a rule of thumb than anything over a few K usually shouldn't go on the stack.

The alternative I presume would be hanging malloced blocks off of dinfo.

Thoughts?

**Comment by Mr-Dave** [comment]
You are right, I didn't consider that and just wanted to make it safe.  I think the solution is really to re-write / eliminate this module to be consistent with the other decompression methods which don't even have these variables.  e.g. netcam_jpeg doesn't use this function and it decompresses images, the picture encoding for the stream doesn't use the encoding function in this module.

**Comment by jogu** [comment]
Yeah, indeed. Should we merge as is then? High stack usage sounds better than unsafe I guess :)

**Comment by Mr-Dave** [comment]
If that is the only issue, I would like to push it to get a functional application for the users with multiple v4l2 cameras.  I'll then open a new issue for tracking an enhancement which we can take a bit more time to resolve regarding what to do with the functions in this module.

**Pull request closed without merging**